### PR TITLE
Fix #8570: Parsing Workflow Gracefully on Deploy

### LINKED ISSUE #8570: Review parse workflow gracefully for supports usage
Based on https://openmetadata.slack.com/archives/C02B6955S4S/p1667854757322749

We are throwing the full pydantic errors instead of catching the right value

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

Fix #8570: Parsing Workflow Gracefully on Deploy

#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py [hunk 1]
```diff
@@ -115,25 +127,57 @@ def build_source(ingestion_pipeline: IngestionPipeline) -> WorkflowSource:
...
-    elif service_type == "mlmodelService":
-        service: MlModelService = metadata.get_by_name(
-            entity=MlModelService, fqn=ingestion_pipeline.service.name
+    entity_class = None
+    try:
+        if service_type == "databaseService":
+            entity_class = DatabaseService
+            service: DatabaseService = metadata.get_by_name(
+                entity=DatabaseService, fqn=ingestion_pipeline.service.name
+            )
```
Comment (TeddyCr) [inline openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py:136]: we can use the `entity_class` for the `entity` argument.

**Review by TeddyCr — CHANGES_REQUESTED** [review]
@ulixius9 just added 1 comment. Could you also share a sample error message?

**Comment by ulixius9** [comment]
Hey @TeddyCr, sure will fix this... also here is the sample error message
```
test_metadata_9jMXOGM0 Failed to deploy Ingestion Pipeline due to airflow API returned Internal Server Error and response {"error": "Internal error while deploying due to [Failed to generate workflow [test_metadata_9jMXOGM0] verify config is correct: We encountered an error parsing the configuration of your SupersetConnection.\n Invalid parameter value for 'hostPort'] "}
```

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py (hunk 1 above)
Comment (pmbrull) [reply openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py]: Hi @TeddyCr @ulixius9 sorry for jumping in. I am not sure I understand the suggestion or the proposed fix 🥲 

Could you please elaborate? The current commit seems more verbose than it has to be.

Thanks!

Comment (TeddyCr) [reply openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py]: There was an `entity_class` added to fetch the suffix for the entity class and provide a better error message.

**Review by TeddyCr — APPROVED** [review]

File: openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py (hunk 1 above)
Comment (pmbrull) [reply openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py]: aaaah got it got it missed the exception later on. Thanks!

**Pull request merged**